Applications/StableDiffusionExample/ContentView.swift

```diff
@@ -146,12 +146,25 @@ actor ModelFactory {
         switch loadState {
         case .idle:
             let task = Task {
-                try await configuration.download { progress in
-                    if progress.fractionCompleted < 0.99 {
-                        reportProgress(
-                            .init(
-                                title: "Download", current: progress.fractionCompleted * 100,
-                                limit: 100))
+                do {
+                    try await configuration.download { progress in
+                        if progress.fractionCompleted < 0.99 {
+                            reportProgress(
+                                .init(
+                                    title: "Download", current: progress.fractionCompleted * 100,
+                                    limit: 100))
+                        }
+                    }
```

Comment:
This is slightly different than the LLM case -- for those the call is "load" and it includes a download.  This is an explicit download step (the weights are _very_ large) and so the detection ends up in the caller (the app in this case).